HandBrake flahub submission

jstebbins opened the pull request.

nedrichards commented:
bot, build fr.handbrake.ghb

nedrichards commented:
other than 'you don't test it on your machine' is there a reason this is x86-64 only?

sr55 commented:
The project as a whole has dropped 32bit builds and isn't something we support anymore.

That said, we were also starting to see increasing problems with memory limitations of 32bit processes.  Particularly with 4K and larger content becoming very common.

jstebbins commented:
In general, the HandBrake team has stopped supporting 32 bit builds.  There are memory limitations on 32 bit platforms that cause failures with certain source file types (e.g. raw 8k video input).  Rather than tell people on 32 bit platforms not to do that, we just dropped 32 bit builds.

In addition, for some time, x265 would only build without asm optimizations on 32 bit.  It's pretty useless without asm optimizations.  I think they fixed that, but we had already dropped 32 bit platforms by the time they fixed it.

TingPing commented on fr.handbrake.ghb.json:
```diff
@@ -0,0 +1,147 @@
+    "command": "ghb", 
+    "runtime-version": "3.28", 
+    "runtime": "org.gnome.Platform", 
+    "sdk": "org.gnome.Sdk"
```
We try to keep modules after all other fields just for readability.

jstebbins replied:
The file is autogenerated.  The order is whatever python json dump writes.

TingPing replied:
If you use an old Python you can put it in an `OrderedDict` (all dicts are ordered in 3.6+)

jstebbins replied:
Thanks for the tip.  I'll look into that.

TingPing commented on fr.handbrake.ghb.json:
```diff
@@ -0,0 +1,147 @@
+                }, 
+                {
+                    "commands": [
+                        "mkdir -p download"
```
Use the `dest` property on sources and you can then avoid this. (It is kinda undefined that paths work in `dest-filename`).

jstebbins replied:
It didn't look like dest-filename would automatically create the destination directory.  Does it?

TingPing replied:
No that is why I'm saying to use `dest` for the subdir in combination with `dest-filename` which is only the filename.

TingPing commented:
Looking at upstream data files:

- You seem to ship a desktop file for two different ids, that doesn't make sense and upstream should pick one. For example the appdata file is broken with old id outside of flatpak.
- You should include an actual release description in appdata, this is exposed to users at update time and is very useful (inside and outside of flatpak)
- I don't think you need to include `<mimetypes>` in appdata file.

Otherwise looks good and none of those are blockers.

jstebbins commented:
The whole desktop application id thing has been a documentation mess.  It looks like the various groups have converged, but there's still plenty of misinformation and counter examples out there.  More counter examples than good examples in fact.  But yes, fixing that is on the todo list.

TingPing commented:
Well there is only one source for the spec and that source recommends the new style: https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#file-naming

jstebbins commented:
Yes.  And 6 months ago, there was no mention of the filename format in that spec.

My point is that when I went seeking examples, there were few good ones to be found.  People are still catching up with the updated spec.

nedrichards commented:
@jstebbins no 32bit is fair enough. Other than annoying broken-ness in h265 do you have anything against aarch64?

eszlari commented on fr.handbrake.ghb.json:
```diff
@@ -0,0 +1,147 @@
+                    "dest-filename": "download/lame-3.100.tar.gz"
+                }, 
+                {
+                    "url": "https://download.handbrake.fr/handbrake/contrib/yasm-1.3.0.tar.gz", 
```
yasm is included in the runtime.

jstebbins replied:
It's installed in a "contrib" tools subdirectory of the HandBrake build directory.  If the HandBrake build directory is not part of the runtime, neither is yasm.

jstebbins replied:
Sorry, i misunderstood your remark.  

I'm not sure why I'm building yasm here.  Perhaps it wasn't in an older version of the runtime (I started experimenting with this nearly 2 years ago). It can be removed in a future release.

eszlari commented on fr.handbrake.ghb.json:
```diff
@@ -0,0 +1,147 @@
+    ], 
+    "command": "ghb", 
+    "runtime-version": "3.28", 
+    "runtime": "org.gnome.Platform", 
```
Afaik handbrake only needs gtk3, so it should be possible to build it against `org.freedesktop.Platform`.

jstebbins replied:
perhaps.  never tried it.

jstebbins replied:
I just tested this.  HandBrake currently uses libnotify which is not included in org.freedesktop.Platform.  I'll have to investigate to see if there is a replacement for this that can be used instead.  But for now, org.gnome.Platform is required.

TingPing replied:
> I'll have to investigate to see if there is a replacement for this that can be used instead.

GNotification is the replacement but that requires you use GApplication which I doubt you do.

Eitherway you may find bundling libnotify to be worth it since the runtime will churn less and users will have less space used.

jstebbins replied:
I do in fact use GApplication, so I'll look into that.  This kind of change is an enhancement rather than a bug fix however, so it won't get updated until we release HandBrake 1.2. 

So in the mean time, I'll look into bundling libnotify.  I would like to push that until our next bug fix release however which would be 1.1.2.

jstebbins replied:
Do you know if there is some trick to getting g_application_send_notification() to work?  This example fails to send any notification on my system 
https://wiki.archlinux.org/index.php/Desktop_notifications#C

TingPing replied:
You have to have a desktop file installed matching the app-id on GNOME-Shell. As a workaround just for testing you can set `env GNOTIFICATION_BACKEND=fdo`

jstebbins replied:
Great.  Thanks.  Got it working.  So 1.2 is set to do away with libnotify.

eszlari commented on fr.handbrake.ghb.json:
```diff
@@ -0,0 +1,147 @@
+{
+    "app-id": "fr.handbrake.ghb", 
+    "modules": [
```
Why don't you add the other dependencies as modules too? This way rebuilding is faster, because already built modules are skipped automatically.

jstebbins replied:
HandBrake's build system already handles building the necessary libraries, including patching,  special configure options and environment variables.  Adding them as modules would be duplication of effort and would require additional testing outside the normal HandBrake tests to make sure it built and runs the same as our regular releases.

eszlari commented on fr.handbrake.ghb.json:
```diff
@@ -0,0 +1,147 @@
+                    "dest-filename": "download/x264-snapshot-20180525-2245.tar.bz2"
+                }, 
+                {
+                    "url": "https://download.handbrake.fr/contrib/x265_2.6.tar.gz", 
```
Is handbrake not compatible with x265 2.8 yet?

jstebbins replied:
This is our 1.1.1 bug fix release.  We do not upgrade contrib libraries in bug fix releases unless there is something very serious that we can not simply add a patch for.  When we release 1.2 we will upgrade this and other libraries.

jstebbins commented:
@nedrichards we prefer not to distribute binaries that we do not test.  We have no idea what may be broken on that platform, nor how it will perform. So I would prefer not to add aarch64.

jstebbins commented on fr.handbrake.ghb.json:
```diff
@@ -0,0 +1,147 @@
+                    "dest-filename": "download/opus-1.2.1.tar.gz"
+                }, 
+                {
+                    "url": "https://download.handbrake.fr/handbrake/contrib/freetype-2.8.1.tar.bz2", 
```
This is a fixable thing in our build system, but I hadn't pursued it yet.  The runtime does not include libass.  And when we build libass (which is optional in our build system) it requires our own locally built version of freetype which requires our locally built harfbuzz. 

It's too late to do anything about it for the 1.1.1 release. I'll update this for the next release.

jstebbins commented on fr.handbrake.ghb.json:
```diff
@@ -0,0 +1,147 @@
+                    "dest-filename": "download/fribidi-0.19.7.tar.gz"
+                }, 
+                {
+                    "url": "https://download.handbrake.fr/handbrake/contrib/harfbuzz-1.7.2.tar.bz2", 
```
See freetype remarks.

nedrichards commented:
@jstebbins makes perfect sense. Feel free to ban anything that you don't want to support.

Similarly - you may wish us to turn off github issues on the repo once accepted and include a readme directing people to your upstream tracker.

jstebbins commented:
@nedrichards that would be helpful.  Easier to track things if it's all in one place.  Thanks for the offer.

TingPing commented on fr.handbrake.ghb.json:
```diff
@@ -0,0 +1,147 @@
+        "--filesystem=~/.config/dconf:ro", 
+        "--talk-name=ca.desrt.dconf", 
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf", 
+        "--filesystem=host"
```
You mention you use `libnotify` so you'll need the `--talk-name=org.freedesktop.Notifications` permission. (See also `flatapk run --log-session-bus ...` to see other things you might be missing).

jstebbins replied:
Thanks

nedrichards commented:
bot, build fr.handbrake.ghb

nedrichards commented:
lgtm. none of those are showstoppers to me.

nedrichards commented:
Repository has been created: https://github.com/flathub/fr.handbrake.ghb

nedrichards commented:
@jstebbins anyone else we should add from upstream?

jstebbins commented:
https://github.com/bradleysepos would like to be added.  Waiting on responses from others.  Will let you know.

jstebbins commented:
And thanks to everyone for you help and input :grinning:

jstebbins commented:
@sr55 is also interested

TingPing commented:
@jstebbins Added both.

sr55 commented:
Thanks :)
